### ISSUE TITLE
vertical/visual block shift should apply only from the point where the block started, not from the beginning of line

### ISSUE THREAD

**Issue by wafflette** [vrapper/vrapper#612]
currently if you select a vertical block (Ctrl-v command in vim), and use >> command to shift it, vrapper will shift the entire line instead of starting at the beginning of block.

**Comment by keforbes** [comment]
I don't know if I should call this a feature request or a defect.  It looks like we never bothered to add a case for visual block mode in `InsertShiftWidth.java`.

**Comment by albertdev** [comment]
@keforbes
I'm going to split out the block code to a new class. My main concern is figuring out how `softtabstop` reacts with blocks aligned on e.g 3 spaces...

**Comment by albertdev** [comment]
It turns out that this is tricky code...

Shifting right needs to deal with tabs which might be  1-8 spaces long, shifting left needs to know how to "borrow" spaces from a tab without changing the rest of the line. I'll need some time to write and test this.

**Comment by albertdev** [comment]
There we go, this should now work correctly in the next release and unstable build.

**Comment by keforbes** [comment]
I've updated the unstable update site with a new build (0.53.20150318) which includes this fix.  Please give it a try and let us know if you run into any other problems.  Thanks!

**Comment by wafflette** [comment]
Mostly works now however if you include empty space, ie a shorter line into visual block, very strange things happen, for instance if you edit this with a vertical block starting at the first S and including 3 lines, in eclipse the vertical block will show as starting at S and ending past the b's. So there's actually a discrepancy between vim and vrapper in how it shows the visual blocks. I think vrapper vertical blocks are actually better than vims but it should simply ignore lines that are past their length.

aaaaaaaaaSaaaaaaaaa
aaaaaaaaaSaaaaaaaaa
bbb

**Comment by albertdev** [comment]
No, what it should do is draw where the caret is at and make sure the block includes everything up to that character the way Vim is doing. That visual issue is actually a bug we're aware of, but it will take some work to properly fix it.

I also noticed that it actually did select to where the caret is, it only doesn't show that. As a result, I triggered some other bug where the insert shiftwidth does some weird stuff and messes up the undo history. Of course I'm going to fix that one before the next release.

For now, simply don't move into such short lines - jumping to the one after that seems to work as intended. Take for example the following:

````
aaaaaaaaaSaaaaaaaaa
aaaaaaaaaSaaaaaaaaa
bbb
aaaaaaaaaSaaaaaaaaa
````

Doing a shift operation should not touch the 3rd line and still change all others.
